### PR TITLE
Fix three issues 

### PR DESCRIPTION
**Pull request by 5SMNOONMS5**
1.https://github.com/Karambirov/GitHubSearch/issues/47
2.https://github.com/Karambirov/GitHubSearch/issues/43
3.https://github.com/Karambirov/GitHubSearch/issues/46

### CODE REVIEW

File: GitHubSearch/Modules/Controllers/Search/SearchViewController.swift [hunk 1]
```diff
@@ -93,7 +93,7 @@ extension SearchViewController {
         searchController.obscuresBackgroundDuringPresentation = false
         searchController.dimsBackgroundDuringPresentation     = false
         searchController.hidesNavigationBarDuringPresentation = false
-        searchController.definesPresentationContext           = true
```
Comment (Karambirov) [inline GitHubSearch/Modules/Controllers/Search/SearchViewController.swift]: Why did you delete `searchController.`?

File: GitHubSearch/Services/RepositoryService.swift [hunk 2]
```diff
@@ -33,12 +33,14 @@ final class RepositoryService {
...
 
     func toggleFavorite(_ repository: Repository) {
-        repository.isFavorite.toggle()
+        realmService?.update {
+            repository.isFavorite = !repository.isFavorite
+        }
+        
         if repository.isFavorite {
-            realmService?.save(repository)
+            self.realmService?.save(repository)
```
Comment (Karambirov) [inline GitHubSearch/Services/RepositoryService.swift]: Why do you use `self`? It's not a closure.

Comment (5SMNOONMS5) [reply GitHubSearch/Services/RepositoryService.swift]: You can delete it later, sorry i didn't notice that.

Comment (Karambirov) [reply GitHubSearch/Services/RepositoryService.swift]: Can you do it yourself?

**Comment by Karambirov** [comment]
Close #47 #43 #46

File: GitHubSearch/Modules/Controllers/Search/SearchViewController.swift (hunk 1 above)
Comment (5SMNOONMS5) [reply GitHubSearch/Modules/Controllers/Search/SearchViewController.swift]: I change from ```searchController.definesPresentationContext``` to ```self``` which is ur ```SearchViewController```.
And for my understanding and from the documentation [definesPresentationContext](https://developer.apple.com/documentation/uikit/uiviewcontroller#//apple_ref/occ/instp/UIViewController/definesPresentationContext)
you should set ```definesPresentationContext = true``` to tell ur ```SearchViewController``` view will be covered.

File: GitHubSearch/Services/RepositoryService.swift (hunk 2 above)
Comment (5SMNOONMS5) [reply GitHubSearch/Services/RepositoryService.swift]: ok i will make another pr

File: GitHubSearch/Modules/Controllers/Search/SearchViewController.swift (hunk 1 above)
Comment (5SMNOONMS5) [reply GitHubSearch/Modules/Controllers/Search/SearchViewController.swift]: And I found out [here](https://thomasdenney.co.uk/blog/2014/10/5/uisearchcontroller-and-definespresentationcontext/). plz check it.
